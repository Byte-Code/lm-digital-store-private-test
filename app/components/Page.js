import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';

const PageWrapper = styled.div`
  background: ${props => props.background};
  padding: ${props => props.padding};
  height: ${props => props.height};
  min-height: 1920px;
`;

export default class Page extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    background: PropTypes.string,
    height: PropTypes.string,
    padding: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  }

  static defaultProps = {
    background: '#ffffff',
    padding: 0,
    height: 'auto'
  }

  render() {
    const { children, background, padding, height } = this.props;

    return (
      <PageWrapper
        background={background}
        padding={padding}
        height={height}
      >
        {children}
      </PageWrapper>
    );
  }
}
