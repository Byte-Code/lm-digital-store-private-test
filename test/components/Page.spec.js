import React from 'react';
import { shallow } from 'enzyme';

import Page from '../../app/components/Page';

describe('Page', () => {
  it('should render properly', () => {
    const result = shallow(<Page><p>hello</p></Page>);
    expect(result).toMatchSnapshot();
  });
});
