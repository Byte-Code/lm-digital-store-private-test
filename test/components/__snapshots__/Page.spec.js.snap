// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render properly 1`] = `
<styled.div
  background="#ffffff"
  height="auto"
  padding={0}
>
  <p>
    hello
  </p>
</styled.div>
`;
